### PR TITLE
[TOPIC-BLE-LLCP] Continued work on LLCP

### PR DESCRIPTION
**Pull request by thoh-ot**

### CODE REVIEW

File: tests/bluetooth/ctrl_llcp/src/main.c [hunk 1]
```diff
@@ -1511,25 +1511,38 @@ void test_api_remote_encryption_start_no_ltk(void)
...
+			);
+
+	ztest_test_suite(public,
+			 ztest_unit_test(test_api_init),
 			 ztest_unit_test(test_api_connect),
-			 ztest_unit_test(test_api_disconnect),
+			 ztest_unit_test(test_api_disconnect)
+			);
+
+	ztest_test_suite(version_exchange,
```
Comment (kruithofa) [inline tests/bluetooth/ctrl_llcp/src/main.c]: Eventually we should move these tests to their own test-suite/module/files

File: tests/bluetooth/ctrl_llcp/src/main.c [hunk 2]
```diff
@@ -464,7 +464,9 @@ void lt_tx(helper_pdu_opcode_t opcode, struct ull_cp_conn *conn, void *param)
 	zassert_equal(0, ret, "k_mem_slab_alloc failed\n");
 
 	pdu = (struct pdu_data *) rx->pdu;
-	helper_pdu_encode[opcode](pdu, param);
+	if (helper_pdu_encode[opcode]) {
```
Comment (kruithofa) [inline tests/bluetooth/ctrl_llcp/src/main.c]: Should we make this into a zassert? If we come here we want to send a PDU, so we should fill in data in this PDU

File: tests/bluetooth/ctrl_llcp/src/main.c [hunk 3]
```diff
@@ -431,11 +431,11 @@ typedef enum {
 	LL_ENC_RSP,
 	LL_START_ENC_REQ,
 	LL_START_ENC_RSP,
-} helper_opcode_t;
+} helper_pdu_opcode_t;
```
Comment (kruithofa) [inline tests/bluetooth/ctrl_llcp/src/main.c]: Note extra comma at end of previous line

Comment (thoh-ot) [reply tests/bluetooth/ctrl_llcp/src/main.c]: Common construct found all over the code base

File: tests/bluetooth/ctrl_llcp/src/main.c (hunk 2 above)
Comment (thoh-ot) [reply tests/bluetooth/ctrl_llcp/src/main.c]: I can do that

File: tests/bluetooth/ctrl_llcp/src/main.c (hunk 1 above)
Comment (thoh-ot) [reply tests/bluetooth/ctrl_llcp/src/main.c]: Agree

File: tests/bluetooth/ctrl_llcp/src/main.c [hunk 4]
```diff
@@ -510,6 +515,25 @@ void ut_rx_q_is_empty()
 	zassert_is_null(ntf, NULL);
 }
 
+void prepare(struct ull_cp_conn *conn)
```
Comment (kruithofa) [inline tests/bluetooth/ctrl_llcp/src/main.c:523]: Rename to something more descriptive, like event_prepare

**Review by kruithofa — APPROVED** [review]

**Pull request merged**